python/mp/effects/bin_demo.py

```diff
@@ -0,0 +1,36 @@
+import copy
+from mp import color as _color
+from mp.effects import effect, bin, looper, sparkle
+
+class BinDemo(effect.Effect):
+    """
+    A demonstration of a composite effect.
+    """
+
+    # Wish there were a better way than requiring this every time
+    dm = copy.deepcopy(effect.Effect.dm)
+
+    def __init__(self, bead_set, color=_color.Color(), **kwargs):
+        super().__init__(name="bin_demo", bead_set=bead_set, color=color, **kwargs)
+
+        self.count = 0
+        self.bin = bin.Bin(bead_set, rosary=self.rosary)
```

Comment:
make a new Bin, assign it to self.bin. Note that you must pass rosary=self.rosary! Sometimes I wonder if it wouldn't be so bad to have just a single global Rosary that all effects refer to, rather than making sure it is set in each effect.